linux/Platform.c

```diff
@@ -1085,3 +1091,34 @@ void Platform_done(void) {
    LibSensors_cleanup();
 #endif
 }
+
+#ifndef BUILD_STATIC
+
+static void* libsystemd_handle;
+static size_t libsystemd_loaded;
+
+void* Platform_load_libsystemd(void) {
```

Comment:
I write some ideas just in case you are planning to refactor this further:
1. The LibSystemd init code can become its own C files, `LibSystemd.c` and `LibSystemd.h`. Within it the `LibSystemd_init()` function would perform what you have `Platform_load_libsystemd()` here. Look at the existing `linux/LibSensors.c` file for what I mean. Dynamically loaded libraries should have the init and unload functions like that.
2. When a libsystemd handle has been `dlopen`'d, all the symbols htop requires should resolve all at once. No need to have the `resolve()` code in different places for the same library.
3. Make the resolved function pointers publicly assessible. That is, put these in `linux/LibSystemd.h` when needed:

```c
// In "linux/LibSystemd.h"
#ifdef BUILD_STATIC

#define sym_sd_bus_open_system sd_bus_open_system
#define sym_sd_bus_open_user sd_bus_open_user
#define sym_sd_bus_get_property_string sd_bus_get_property_string
#define sym_sd_bus_get_property_trivial sd_bus_get_prope
rty_trivial
#define sym_sd_bus_unref sd_bus_unref
#define sym_sd_get_sessions sd_get_sessions

#else

extern int (*sym_sd_bus_open_system)(sd_bus**);
extern int (*sym_sd_bus_open_user)(sd_bus**);
extern int (*sym_sd_bus_get_property_string)(sd_bus*, const char*, const char*, const char*, const char*, sd_bus_error*, char**);
extern int (*sym_sd_bus_get_property_trivial)(sd_bus*, const char*, const char*, const char*, const char*, sd_bus_error*, char, void*);
extern sd_bus* (*sym_sd_bus_unref)(sd_bus*);
extern int (*sym_sd_get_sessions)(char***);

#endif